lib/profile/commands/apply.rb

```diff
@@ -173,6 +177,30 @@ def check_nodes_exist(names=[])
           raise out
         end
       end
+      
+      def expand_brackets(str)
+        labels = []
+        contents = str[/\[.*\]/] 
+        if contents.nil?
+          labels = [str]
+        else
+          left = str[/[^\[]*/]
+          right = str[/].*/][1..-1]
+          if contents.match(/^\[[0-9]+-[0-9]+\]$/)
+            nums = contents[1..-2].split("-")
+            if nums.first.to_i < nums.last.to_i
+              (nums.first..nums.last).each do |index|
+                labels.append(left + index + right)
+              end
+            else
+              raise "Invalid range, ensure that the end index is greater than the start index"
```

Comment:
To avoid nested if statements getting a bit too deep, could you separate this check (and, ideally, the `[START-END]` check too) into its own block please?

```ruby
unless nums.first.to_i < nums.last
  raise "Invalid range[...]"
end

nums = contents[1..-2].split("-")
...
...
```